### DISCUSSION TITLE
multiple key exchange

### DISCUSSION THREAD

**Discussion by keskinadem** [strongswan/strongswan#1561] [Q&A] [answered]
What is the planned date that version of 6.0.0 is going to be stabily released? And do you have multiple key exchange Sample Test Cases like https://www.strongswan.org/testing/testresults/ikev2/?

**Comment by tobiasbrunner** [comment, marked as answer]
> What is the planned date that version of 6.0.0 is going to be stabily released?

There is currently no ETA.

> And do you have multiple key exchange Sample Test Cases like https://www.strongswan.org/testing/testresults/ikev2/?

There is the [`ikev2/rw-cert-qske` scenario](https://www.strongswan.org/testing/testresults6dr/ikev2/rw-cert-qske/) in the 6.0 beta test results. And there is also a [Docker setup with liboqs](https://github.com/strongX509/docker/tree/master/pq-strongswan) for some quick tests.

**Comment by Gor1234567** [comment]
I'm a little confused,
Could you please explain - strongSwan6(beta) is done according to [https://wiki.strongswan.org/projects/strongswan/wiki/NTRU] or [https://datatracker.ietf.org/doc/rfc9370/] ? I mean how are quantum-resistant algorithms will be used in 6 version?

**Reply by tobiasbrunner** [reply]
Check the link to the Docker setup above.

**Reply by Gor1234567** [reply]
Thanks, it helped a lot. I still have 2 questions regarding strongSwan release and multiple key exchange.

1) How far is beta5 from release version?

2) Implementation of additional key exchanges is similar to the _following KEM interpretation_ or not:
_The IKE initiator generates a random ephemeral public/private key pair for the specified security strength.
The IKE initiator sends the public key in the KEi payload to the IKE responder.
The IKE responder generates a random secret s with a size of twice the security strength and encrypts it with the public key.
The IKE responder sends the encrypted secret in the KEr payload to the IKE initiator
The IKE initiator decrypts the KEr payload using the private key and extracts the secret s.
With IKEv2 both initiator and responder use the secret s to compute SKEYSEED._
